client/test/integration/markLogicIntegrationTestHelper.ts

```diff
@@ -174,6 +176,14 @@ export class IntegrationTestHelper {
             await JsDebugManager.disconnectFromNamedJsDebugServer(globalThis.integrationTestHelper.attachServerName);
             globalThis.integrationTestHelper.attachedToServer = false;
         }
+        // Some tests can leave the MarkLogic server in a state that causes future tests to fail.
+        // The only reliable method for resolving this state is to restart the MarkLogic server.
+        if (globalThis.integrationTestHelper.restartServer) {
+            await this.restartMarkLogic();
+            globalThis.integrationTestHelper.restartServer = false;
+            await wait(5000);
```

Comment:
Is the hope that 5s is long enough for MarkLogic to restart? The /admin/v1/timestamp endpoint can be used in a loop instead to reliably wait for MarkLogic to restart, that's what ml-app-deployer does.